client/my-sites/customer-home/main.jsx

```diff
@@ -0,0 +1,254 @@
+import QueryJetpackPlugins from 'components/data/query-jetpack-plugins';
+import Main from 'components/main';
+import VerticalNav from 'components/vertical-nav';
+import VerticalNavItem from 'components/vertical-nav/item';
+import { preventWidows } from 'lib/formatting';
+import SidebarNavigation from 'my-sites/sidebar-navigation';
+import { getSelectedSite, getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
+import {
+	isJetpackSite,
+	getCustomizerUrl,
+	canCurrentUserUseCustomerHome,
+} from 'state/sites/selectors';
+import isSiteAutomatedTransfer from 'state/selectors/is-site-automated-transfer';
+import PageViewTracker from 'lib/analytics/page-view-tracker';
+import DocumentHead from 'components/data/document-head';
+import getSiteChecklist from 'state/selectors/get-site-checklist';
+import QuerySiteChecklist from 'components/data/query-site-checklist';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+const ActionBox = ( { action, iconSrc, label } ) => {
+	const buttonAction = 'function' === typeof action ? { onClick: action } : { href: action };
+	return (
+		<div className="customer-home__box-action">
+			<Button { ...buttonAction }>
+				<img src={ iconSrc } alt="" />
+				<span>{ label }</span>
+			</Button>
+		</div>
+	);
+};
+
+class Home extends Component {
+	static propTypes = {
+		checklistMode: PropTypes.string,
+
+		site: PropTypes.object.isRequired,
+		siteId: PropTypes.number.isRequired,
+		siteSlug: PropTypes.string.isRequired,
+		customizeUrl: PropTypes.string.isRequired,
+		canUserUseCustomerHome: PropTypes.bool.isRequired,
+		siteChecklist: PropTypes.object.isRequired,
+		isJetpackNotAtomic: PropTypes.bool.isRequired,
+	};
+
+	render() {
+		const { translate, canUserUseCustomerHome } = this.props;
+
+		if ( ! canUserUseCustomerHome ) {
+			return (
+				<EmptyContent
+					title={ preventWidows( translate( 'You are not authorized to view this page.' ) ) }
+					line={ preventWidows(
+						translate(
+							"Ensure you have selected the right site on the sidebar. If it's correct, you might want to contact the administrator of this site."
+						)
+					) }
+					illustration="/calypso/images/illustrations/error.svg"
+				/>
+			);
+		}
+
+		const { siteId, checklistMode, siteChecklist, isJetpackNotAtomic } = this.props;
+
+		const checklistToRender = isJetpackNotAtomic ? (
+			<Fragment>
+				<QueryJetpackPlugins siteIds={ [ siteId ] } />
+				<JetpackChecklist />
+			</Fragment>
+		) : (
+			<ChecklistWpcom displayMode={ checklistMode } />
+		);
+
+		return (
+			<Main className="customer-home__main is-wide-layout">
+				<PageViewTracker path={ `/home/:site` } title={ translate( 'Customer Home' ) } />
+				<DocumentHead title={ translate( 'Customer Home' ) } />
+				<SidebarNavigation />
+				{ siteId && isEmpty( siteChecklist ) && <QuerySiteChecklist siteId={ siteId } /> }
+				{ siteId &&
+					! isEmpty( siteChecklist ) &&
```

Comment:
Since you're checking `isEmpty( siteChecklist )` twice, maybe you could assign that value to a variable. 

Also, it looks like we're importing two lodash methods, `isEmpty` and `some` for operations we could do with plain JS, especially since you're already assigning `{}` to `siteChecklist` in the connect method.

```
const  hasChecklistTasks = siteChecklist && Array. isArray( siteChecklist.tasks );

....

				{ siteId && ! hasChecklistTasks && <QuerySiteChecklist siteId={ siteId } /> }
                { siteId
                    && hasChecklistTasks
                    && siteChecklist.tasks.some( task => task. isCompleted === false )
                    ? checklistToRender
                    : this.renderCustomerHome()
                }  
```

Alternatively, you could pass this stuff as props via connect. Might make unit testing the UI easier too.

```
    const hasChecklistTasks: siteChecklist !== null && Array. isArray( siteChecklist.tasks )
	return {
       ....
        hasChecklistTasks,
        shouldRenderChecklist: siteId && hasChecklistTasks && siteChecklist.tasks.some( task => task. isCompleted === false ),
	};
```

As always, feel free to ignore this! :) Looking forward to seeing the home live in production.

Reply:
Thanks for the suggestion! I added props to connect, although with a slight change. Didn't check for siteId because I realized it's enough to know if siteChecklist is null. I also removed the siteId check in the render method for the same reason, specially now that shouldRenderChecklist encompasses everything.
Thanks again!
```js
shouldRenderChecklist: hasChecklistTasks && ...
```